src/aiida/cmdline/commands/cmd_daemon.py

```diff
@@ -217,24 +218,27 @@ def stop(ctx, no_wait, all_profiles, timeout):
 def restart(ctx, reset, no_wait, timeout):
     """Restart the daemon.
 
-    By default will only reset the workers of the running daemon. After the restart the same amount of workers will be
-    running. If the `--reset` flag is passed, however, the full daemon will be stopped and restarted with the default
+    The daemon is stopped before being restarted with the default
     number of workers that is started when calling `verdi daemon start` manually.
 
     Returns exit code 0 if the result is OK, non-zero if there was an error.
     """
     if reset:
-        # These two lines can be simplified to `ctx.invoke(start)` once issue #950 in `click` is resolved.
-        # Due to that bug, the `callback` of the `number` argument the `start` command is not being called, which is
-        # responsible for settting the default value, which causes `None` to be passed and that triggers an exception.
-        # As a temporary workaround, we fetch the default here manually and pass that in explicitly.
-        number = ctx.obj.config.get_option('daemon.default_workers', ctx.obj.profile.name)
-        ctx.invoke(stop)
-        ctx.invoke(start, number=number)
-        return
-
-    echo.echo('Restarting the daemon... ', nl=False)
-    execute_client_command('restart_daemon', wait=not no_wait, timeout=timeout)
+        warn_deprecation(
+            '`--reset` flag is deprecated. Now, `verdi daemon restart` by default restarts the full daemon.', version=3
+        )
+    if no_wait:
+        warn_deprecation('The `--no-wait` flag is deprecated and no longer has any effect.', version=3)
```

Comment:
I think using `echo.echo_deprecated` might be nicer for CLI options?

Reply:
Could use that as well. That function doesn't go through Python's warning system though but simply prints a message to the logger. This means it doesn't respect the `showdeprecationwarnings` option. Maybe it is better to have it respect that config option, rather than having slightly nicer formatting? For me it is not that big of a problem and would be fine with both tbh

Reply:
Agreed about not being super important, but since this is in the CLI, I think nicer output would be preferred. :shrug: And complying with the deprecation means less typing so I don't think a potential user that wants to ignore all warnings will be mad about it. :-)

Reply:
Ok fine by me, @khsrali please replace `warn_deprecation` with `aiida.cmdline.utils.echo.echo_deprecated`

Reply:
Alright, thanks @danielhollas for your review. Replaced.